common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -512,7 +514,9 @@ private static boolean eligibleToCache(CommandResult commandResult) {
      * @return
      */
     private static boolean eligibleToCacheException(BaseException exception) {
-        if (exception instanceof IntuneAppProtectionPolicyRequiredException) {
+        //TODO : ADO 1373343 Add the whole transient exception category.
+        if (exception instanceof IntuneAppProtectionPolicyRequiredException
+                || ErrorStrings.DEVICE_NETWORK_NOT_AVAILABLE.equals(((ClientException)exception).getErrorCode())) {
```

Comment:
you probably want to check that it is instance of `ClientException` before you cast

Reply:
(No cast is required here)

Reply:
Looks like you have already got that feedback

Reply:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1329#discussion_r619458892